### PR TITLE
chore: update tsconfig to target es2020

### PR DESCRIPTION
**Pull request by rwaldron**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
     "compilerOptions": {
-        "target": "es2015",
-        "module": "es2015",
+        "target": "es2020",
+        "module": "es2020",
         "moduleResolution": "node",
         "esModuleInterop": true,
-        "lib": ["es2015", "es2016", "es2017", "es2018", "DOM"],
+        "lib": ["es2015", "es2016", "es2017", "es2018", "es2019", "es2020", "DOM"],
```
Comment (jdalton) [inline tsconfig.json:7]: The array in locker looks different. Lets copy pasta it:
```js
["dom", "es2015", "es2016", "es2017", "es2018", "es2019", "es2020"]
```

**Review by garychangsf — APPROVED** [review]

File: tsconfig.json [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
         "declaration": false,
         // Enhance Strictness
         "strict": true,
```
Comment (jdalton) [inline tsconfig.json:10]: Also noticed this is a more opinionated compiler options config overall than Locker. I wonder if Locker could benefit from more of this.

**Review by jdalton — COMMENTED** [review]
Minor nit on the ordering but other than that it's good to go.

**Review by jdalton — APPROVED** [review]

File: tsconfig.json (hunk 1 above)
Comment (rwaldron) [reply tsconfig.json:7]: Good call

File: tsconfig.json (hunk 2 above)
Comment (manuel-jasso) [reply tsconfig.json:10]: +1 Even though locker and near-membrane are separate repos, we should strive to keep them the same in terms of dependencies, configs, etc.

**Pull request merged**